pkg/dmap-funcs/rate-limiter/global-token-bucket/global-token-bucket.go

```diff
@@ -105,14 +148,46 @@ func (gtb *GlobalTokenBucket) Close() error {
 	if err != nil {
 		return err
 	}
+	err = gtb.jobGroup.DeregisterJob(gtb.name)
+	if err != nil {
+		return err
+	}
 	return nil
 }
 
 func (gtb *GlobalTokenBucket) executeTakeRequest(ctx context.Context, label string, n float64, canWait bool, deadline time.Time) (bool, time.Duration, float64, float64) {
+	getWaitTime := func(availableAt time.Time) time.Duration {
+		var waitTime time.Duration
+		if !availableAt.IsZero() {
+			waitTime = time.Until(availableAt)
+			if waitTime < 0 {
+				waitTime = 0
+			}
+		}
+		return waitTime
+	}
+
 	if gtb.GetPassThrough() {
 		return true, 0, 0, 0
 	}
 
+	c := &counter{}
+
+	if !canWait {
+		existing, loaded := gtb.counters.LoadOrStore(label, c)
+		if loaded {
+			c = existing.(*counter)
+		}
+
+		c.lock.Lock()
```

Comment:
This lock will serialize calls to olric for one token bucket. Feel we should not serialize unless we already over the limit

Reply:
Instead of LoadOrStore, simply check if counter exists in cache. Call store only after making the dmap call.